lib/addons/cloud-watch-insights/index.ts

```diff
@@ -75,11 +67,12 @@ export class CloudWatchInsights extends CoreAddOn {
       insightsSA.role.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('CloudWatchAgentServerPolicy'));
       insightsSA.role.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AWSXrayWriteOnlyAccess'));
 
-      const insightsAddon = new eks.CfnAddon(context.scope,  "CloudWatchInsightsAddon", {
+      const insightsAddon = new eks.CfnAddon(clusterInfo.cluster.stack,  "CloudWatchInsightsAddon", {
         addonName: defaultProps.addOnName,
```

Comment:
lets not do this. Lets use the `deploy` method of the `coreAddon` so we dont lose the functionality of `deploy` method of `coreAddon` which has features like `provideVersion` etc. With this approach if `auto` is passed, `auto` is passed as version to addon creation.